src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -453,7 +447,11 @@ fn run_tool_check_step(
         target,
         builder.kind,
         path,
-        source_type,
+        // Currently, all of the tools that use this macro/function are in-tree.
+        // If support for out-of-tree tools is re-added in the future, those
+        // steps should probably be marked non-default so that the default
+        // checks aren't affected by toolstate being broken.
+        SourceType::InTree,
```

Comment:
@rust-lang/bootstrap do you know if there's a specific reason we support out-of-tree tools? Does idk distros or some other consumer rely on this support?

Reply:
It looks like the general concept of `SourceType` was introduced in #73297, to control whether warnings are denied by default or not.

At that time, none of the uses of `tool_check_step!` specified a source type other than `InTree`. So it's plausible that this macro has never actually been used with out-of-tree tools.

Reply:
Also note that this PR only removes explicit `SourceType` support from one particular macro. If an “out of tree” tool hypothetically needed to be added, it shouldn't be much trouble to either adjust this macro, or declare an alternative step or macro.

Reply:
Yeah, I'm in favor of removing this unused code. If we need support for non-`InTree` tools, we can add it back when the need arises.